File: 682-baseball-game/baseball-game.js
/**
 * @param {string[]} operations
 * @return {number}
 */
var calPoints = function (operations) {
    let stack = [];

    for (let ops of operations) {
        if (ops == '+') {
            stack.push(stack[stack.length - 1] + stack[stack.length - 2]);
        }
        else if (ops == 'D') {
            stack.push(2 * stack[stack.length - 1]);
        }
        else if (ops == 'C') {
            stack.pop();
        }
        else {
            stack.push(Number(ops))
        }
    }
    return stack.reduce((a, b) => a + b, 0);
};

File: 682-baseball-game/baseball-game.java
class Solution {
    public int calPoints(String[] operations) {
        Stack<Integer> score = new Stack<>();

        for(String s : operations) {
            if (s.equals("+")) {
                int top = score.pop();
                int secondTop = score.peek();
                score.push(top);
                score.push(top + secondTop);
            } else if (s.equals("D")) {
                int prev = score.peek();
                score.push(prev * 2);
            } else if (s.equals("C")) {
                score.pop();
            } else {
                score.push(Integer.parseInt(s));
            }
        }

        int sum = 0;
        for (int num : score) {
            sum += num;
        }

        return sum;
    }
}